### DISCUSSION TITLE
Any GCP Compute Engine providers in development?

### DISCUSSION THREAD

**Discussion by Hdom** [cloudbase/garm#192] [Q&A] [answered]
Hello, I am wondering if there is anyone out the working on a provider for GCP Compute Engine.

**Comment by alexpilotti** [comment]
Hi @Hdom, we’re currently working on adding new providers, the GCE one is planned for the end of January

**Comment by fabi200123** [comment, marked as answer]
Hi @Hdom, you can check out the provider for GCP Compute Engine [here](https://github.com/cloudbase/garm-provider-gcp). Let me know if you have any questions related to it!

**Reply by Hdom** [reply]
Hey @fabi200123, while we waited for the official one we created our own GCP provider, I will look into switching to use this one but it seems to be missing some crucial features, like setting custom labels and tags, and using snapshots as the source disk. Are there any plans to support these things?

**Reply by gabriel-samfira** [reply]
Hi @Hdom !

That actually sounds amazing! Is your provider open source? Community providers are always welcome. They don't need to be "official" or written in Go. For example, the k8s provider is written and maintained by the amazing folks at Mercedes-Benz.

If you prefer to not maintain or open source it, that's also fine. It would however be really appreciated if you open a few issues against the one @fabi200123 linked to, describing in detail the needs you have (like the things you mentioned here), and we'll try to add those features.

**Reply by Hdom** [reply]
@gabriel-samfira Unfortunately I was not allowed to share the provider, I will look into contributing, we used the openstack provider as a base and made a lot of assumptions, some trial and error and comparisons to the open source GCP provider got us the rest of the way. 

The only difference at this point is the addition of some custom values in our final spec, we use labels for billing and network policy filtering, and network tags to enable ssh, rdp, etc.

**Reply by gabriel-samfira** [reply]
That's perfectly fine. Could you [open an issue](https://github.com/cloudbase/garm-provider-gcp/issues/new) for every feature you're missing? We will try to implement those and you can simply switch to the open source provider.

**Reply by Hdom** [reply]
I apologize for the delay, here are the 3 issues from our previous discussion on Labels, Tags, and Snapshots:
https://github.com/cloudbase/garm-provider-gcp/issues/3
https://github.com/cloudbase/garm-provider-gcp/issues/4
https://github.com/cloudbase/garm-provider-gcp/issues/5
